ibridges/session.py

```diff
@@ -106,8 +108,36 @@ def has_valid_irods_session(self) -> bool:
         """
         return self.irods_session is not None and self.server_version != ()
 
+    def _network_check(self, hostname: str, port: int) -> bool:
```

Comment:
Can we check the network only in case of trouble? Or is it something we always have to do?

Reply:
I only found a way in python to check whether it is possible to establish a connection to a server through a certain port. That is what this function implements.
The reasons for failing can be: 1) no internet, 2) wrong server name, 3) wrong port.

Reply:
I understand, but can we check the network after we get an error somewhere? It would save some time (I know not a lot probably) for creating the session, and we would not have an issue if somehow this network check fails, but it could actually successfully connect (I don't know if this is possible, but perhaps with a fancy network setup?).